### DISCUSSION TITLE
What is the difference between service and upstream?

### DISCUSSION THREAD

**Discussion by xlb0479** [apache/apisix#5010] [Q&A]
These two may have difference in concept but their usage almost the same.
Any help?

**Comment by tzssangglass** [comment]
> but their usage almost the same.

there are totally different.

pls read docs first, https://github.com/apache/apisix/blob/master/docs/en/latest/architecture-design/upstream.md and https://github.com/apache/apisix/blob/master/docs/en/latest/architecture-design/service.md

**Reply by xlb0479** [reply]
Yes I have read these docs already.
I can figure out the difference between service and upstream from the json request body.
But when I use the dashboard to setup a service or a upstream, the ui confused me alot, these two things are almost the same.

**Reply by tzssangglass** [reply]
You can list some details of where the confusion lies

**Reply by juzhiyuan** [reply]
`Upstream` is the Backend Server (or targeting nodes), `Service` is bound with a bundle of `Route(s)/APIs`. 

> the ui confused me alot, these two things are almost the same.

Please take a screenshot and have a more clear description, which part make you confused 🤔
